deploy-service/common/src/main/java/com/pinterest/deployservice/db/DBEnvironDAOImpl.java

```diff
@@ -47,6 +48,8 @@ public class DBEnvironDAOImpl implements EnvironDAO {
         "UPDATE environs SET external_id=? WHERE env_name=? AND stage_name=?";
     private static final String GET_ENV_BY_ID =
         "SELECT * FROM environs WHERE env_id=?";
+    private static final String GET_ENV_BY_DEPLOY_ID =
+        "SELECT e.* FROM environs e INNER JOIN deploys d ON e.env_id = d.env_id WHERE d.deploy_id=?";
```

Comment:
Thanks for being detail oriented. I compared the 2, since the key used are primary, we won't have performance impact. 

## When the deploy exists

### Current

```mysql
explain SELECT e.* FROM environs e INNER JOIN deploys d ON e.env_id = d.env_id WHERE d.deploy_id="0--4nnYXQMi7bXh-mbzVpA";
+----+-------------+-------+------------+-------+------------------------+---------+---------+-------+------+----------+-------+
| id | select_type | table | partitions | type  | possible_keys          | key     | key_len | ref   | rows | filtered | Extra |
+----+-------------+-------+------------+-------+------------------------+---------+---------+-------+------+----------+-------+
|  1 | SIMPLE      | d     | NULL       | const | PRIMARY,deploy_env_idx | PRIMARY | 24      | const |    1 |   100.00 | NULL  |
|  1 | SIMPLE      | e     | NULL       | const | PRIMARY                | PRIMARY | 24      | const |    1 |   100.00 | NULL  |
+----+-------------+-------+------------+-------+------------------------+---------+---------+-------+------+----------+-------+
```

### Proposed
```mysql
explain SELECT e.* FROM environs e where e.env_id = (select d.env_id from deploys d WHERE d.deploy_id="0--4nnYXQMi7bXh-mbzVpA");
+----+-------------+-------+------------+-------+---------------+---------+---------+-------+------+----------+-------+
| id | select_type | table | partitions | type  | possible_keys | key     | key_len | ref   | rows | filtered | Extra |
+----+-------------+-------+------------+-------+---------------+---------+---------+-------+------+----------+-------+
|  1 | PRIMARY     | e     | NULL       | const | PRIMARY       | PRIMARY | 24      | const |    1 |   100.00 | NULL  |
|  2 | SUBQUERY    | d     | NULL       | const | PRIMARY       | PRIMARY | 24      | const |    1 |   100.00 | NULL  |
+----+-------------+-------+------------+-------+---------------+---------+---------+-------+------+----------+-------+
```


## When the deploy doesn't exist

### Current
```mysql
explain SELECT e.* FROM environs e INNER JOIN deploys d ON e.env_id = d.env_id WHERE d.deploy_id="123"; 
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+--------------------------------+
| id | select_type | table | partitions | type | possible_keys | key  | key_len | ref  | rows | filtered | Extra                          |
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+--------------------------------+
|  1 | SIMPLE      | NULL  | NULL       | NULL | NULL          | NULL | NULL    | NULL | NULL |     NULL | no matching row in const table |
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+--------------------------------+
```
### Proposed
```mysql
explain SELECT e.* FROM environs e where e.env_id = (select d.env_id from deploys d WHERE d.deploy_id="123");
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+--------------------------------+
| id | select_type | table | partitions | type | possible_keys | key  | key_len | ref  | rows | filtered | Extra                          |
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+--------------------------------+
|  1 | PRIMARY     | NULL  | NULL       | NULL | NULL          | NULL | NULL    | NULL | NULL |     NULL | no matching row in const table |
|  2 | SUBQUERY    | NULL  | NULL       | NULL | NULL          | NULL | NULL    | NULL | NULL |     NULL | no matching row in const table |
+----+-------------+-------+------------+------+---------------+------+---------+------+------+----------+--------------------------------+
2 rows in set (0.00 sec)
```